core/templates/dev/head/domain/exploration/ReadOnlyExplorationBackendApiService.js

```diff
@@ -75,16 +85,19 @@ oppia.factory('ReadOnlyExplorationBackendApiService', [
        * retrieves the exploration from the backend, it will store it in the
        * cache to avoid requests from the backend in further function calls.
        */
-      loadExploration: function(explorationId) {
+      loadExploration: function(explorationId, version) {
         return $q(function(resolve, reject) {
-          if (_isCached(explorationId)) {
+          if (_isCached(explorationId) && !version) {
             if (resolve) {
               resolve(angular.copy(_explorationCache[explorationId]));
             }
           } else {
-            _fetchExploration(explorationId, function(exploration) {
+            _fetchExploration(explorationId, version, function(exploration) {
               // Save the fetched exploration to avoid future fetches.
-              _explorationCache[explorationId] = exploration;
+              // Excludes version for now.
```

Comment:
I love this idea. Also, I was thinking loadVersionedExploration might be a good name for the versioned load method? But on second thought, the name loadExploration that requires an extra parameter seems logical enough. Done. 

Do we want an additional cache for Versioned Exploration's? Or is that overkill for now.